lib/sidekiq/ultimate/use_exists_question_mark.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+require "redis"
+
+module Sidekiq
+  module Ultimate
+    # Redis-rb 4.2.0 renamed `#exists` to `#exists?` and changed behaviour of `#exists` to return integer
+    # https://github.com/redis/redis-rb/blob/master/CHANGELOG.md#420
+    module UseExistsQuestionMark
+      def self.use?
+        return @use if defined?(@use)
+
+        @use = Gem::Version.new(Redis::VERSION) >= Gem::Version.new("4.2.0")
+      end
+    end
```

Comment:
Why not just make a direct convenience function? `Sidekiq::Ultimate.redis_exists?(r, key)`

Reply:
I suggest create a whole new class:

``` ruby
module Sidekiq
  module Ultimate
    class RedisAdapter
      REDIS_420 = Gem::Version.new("4.2.0").freeze

      def initialize(redis)
        @redis = redis
      end

      def exists?(key)
        if REDIS_420 <= @redis.class::VERSION
          @redis.exists?(key)
        else
          @redis.exists(key)
      end

      # more method here
    end
  end
end
```

This way it will be much easier to support Sidekiq 7.0, which uses redis-client instead of redis-rb.

Reply:
Actually there's no need to check for redis client version:

``` ruby
def redis_exists?(redis, key)
  redis.exists(key).then do |value|
    # redis < 4.2:  exists(key) => true|false
    # redis >= 4.2: exists(key) => 1|0
    value == 1 || value == true
  end
end
```

Reply:
I'm going to drop support for redis < 4.2 soon, so decided to keep it simple.